pkg/mimirpb/compat.go

```diff
@@ -315,6 +315,71 @@ func SampleJsoniterDecode(ptr unsafe.Pointer, iter *jsoniter.Iterator) {
 	}
 }
 
+func FromPromCommonToMimirSampleHistogram(src *model.SampleHistogram) *SampleHistogram {
+	return (*SampleHistogram)(unsafe.Pointer(src))
+}
+
+func FromMimirSampleToPromCommonHistogram(src *SampleHistogram) *model.SampleHistogram {
+	return (*model.SampleHistogram)(unsafe.Pointer(src))
+}
+
+// FromFloatHistogramToSampleHistogramProto converts histogram.FloatHistogram to SampleHistogram.
+func FromFloatHistogramToSampleHistogramProto(h *histogram.FloatHistogram) *SampleHistogram {
+	buckets := make([]*HistogramBucket, 0)
```

Comment:
sounds reasonable (currently optimizations - unless to keep float series performance - is very low prio)

> WDYT about initializing the capacity this way? I have mixed feelings because depends on how many empty buckets we have. If there are many, we over allocate, if there are few and we don't do this then we over re-allocate.